zjit/src/hir.rs

```diff
@@ -912,9 +913,23 @@ impl Function {
     fn infer_types(&mut self) {
         // Reset all types
         self.insn_types.fill(types::Empty);
-        for param in &self.blocks[self.entry_block.0].params {
+        let rest_param_idx = if !self.iseq.is_null() && unsafe { get_iseq_flags_has_rest(self.iseq) } {
```

Comment:
To avoid computing this every time we may want to store `param_types` on `Function` or something like that and copy that in instead

Reply:
Sure, I tossed it in a `OnceCell`.